### ISSUE TITLE
Knn doubt and question related to distances between k neigh

### ISSUE THREAD

**Issue by dhorka** [pyg-team/pytorch_geometric#1322]
Hi,

I have some doubts related to the knn implementation: https://pytorch-geometric.readthedocs.io/en/1.3.1/_modules/torch_cluster/knn.html

I see that to find the nearest neighbor, and you're using KDTree from spatial scypi. The first question is related to the backward propagation. As far as I know to use the autograd capabilities from PyTorch you can not use other frameworks, am I right? I suppose that it doesn't matter in this case because knn is only used to find some ind and does not require the grad. Is it correct? The same happens in the Cuda version, you don't need a backpropagation, because of the same reason? It is to confirm that I have understood the behavior correctly. 

I would like to ask if it is possible to add a parameter that returns the distances between the selected neighbors. For instance, you have:

pos=[[0,0,0], [0,0,1], [0,0,3]]

the knn k=1 it will be:
inx = [1,0,1]

I would like to have something like this:
inx = [1,0,1]
dist = [1,1,2]

That is just a toy example. I know that I can do the estimation of the distance after the calculation of knn. However, it would be great to have this because it will save some computation time. Maybe it can be an optional parameter or something like this.

Thanks,

**Comment by rusty1s** [comment]
You are right that those functions do not have a backward implementation since their are only computing indices.. Your requests sounds reasonable, although I am not sure when I will have time to add it. In the end, as you already mentioned, you can compute it afterwards. The additional computation time is negligible.
